From 8df72fb9a914e5a3469685ec6633e24509eedb05 Mon Sep 17 00:00:00 2001
From: PJ <pj@imagine-nyc.com>
Date: Mon, 10 Feb 2020 10:11:41 +0100
Subject: [PATCH] conf: add application/wasm mime type

---
 conf/mime.types | 1 +
 1 file changed, 1 insertion(+)

diff --git a/conf/mime.types b/conf/mime.types
index 2961256..b53f7f7 100644
--- a/conf/mime.types
+++ b/conf/mime.types
@@ -51,6 +51,7 @@ types {
     application/vnd.openxmlformats-officedocument.wordprocessingml.document
                                                      docx;
     application/vnd.wap.wmlc                         wmlc;
+    application/wasm                                 wasm;
     application/x-7z-compressed                      7z;
     application/x-cocoa                              cco;
     application/x-java-archive-diff                  jardiff;
-- 
2.7.1

