diff --git a/pkgs/build-support/rust/default.nix b/pkgs/build-support/rust/default.nix
index 6afe93c7e4c..94caa912f4c 100644
--- a/pkgs/build-support/rust/default.nix
+++ b/pkgs/build-support/rust/default.nix
@@ -86,6 +86,10 @@ in stdenv.mkDerivation (args // {
     ${stdenv.lib.optionalString (stdenv.buildPlatform.config != stdenv.hostPlatform.config) ''
     [target."${stdenv.hostPlatform.config}"]
     "linker" = "${ccForHost}"
+    ${stdenv.lib.optionalString (stdenv.hostPlatform.config == "aarch64-unknown-linux-musl") ''
+    # https://github.com/rust-lang/rust/issues/46651#issuecomment-433611633
+    "rustflags" = [ "-C", "target-feature=+crt-static", "-C", "link-arg=-lgcc" ]
+    ''}
     ''}
     EOF
     cat .cargo/config
