diff --git a/pkgs/build-support/rust/default.nix b/pkgs/build-support/rust/default.nix
index 09802d56d34..ae689d7c8ce 100644
--- a/pkgs/build-support/rust/default.nix
+++ b/pkgs/build-support/rust/default.nix
@@ -95,6 +95,10 @@ stdenv.mkDerivation (args // {
 
   configurePhase = args.configurePhase or ''
     runHook preConfigure
+
+    # https://github.com/NixOS/nixpkgs/issues/61618
+    export HOME=$(mktemp -d)
+
     runHook postConfigure
   '';
 
